src/docs/arch-schema.dox

```diff
@@ -1,11 +1,117 @@
+@ref schema_api. There are several WiredTiger internal objects such as @ref arch-metadata,
+@ref arch-hs, etc. The schema of those objects is locked and cannot be altered from outside
+of WiredTiger.
+
+Schema operations cause an update to the metadata and are performed under the schema lock to avoid
+concurrent operations on the database schema. The following sequence of steps define a generic schema
+operation:
+
+@plantuml_start{schema_generic.png }
+@startuml{schema_generic.png}
+:A schema operation;
+partition with-schema-lock {
+    :perform operation on underlying data-object;
+    :update metadata-file;
+    :checkpoint and sync metadata;
+}
+stop
+@enduml
+@plantuml_end
+
+Apart from the schema API, the schema lock is necessary for many other
+operations in WiredTiger including the following "heavy" database modifications:
+ - The schema lock wraps \c checkpoint \c prepare to avoid any tables being created or dropped
+ during this phase. See @ref arch-checkpoint for details.
+ - \c Rollback \c to \c stable operation acquires the schema lock to make sure no schema changes are
+ done during this complex process. @ref arch-rts has more information on the operation.
+ - A \c backup \c cursor also holds the schema lock because it must guarantee a consistent view
+ of what files and tables exist while it is being used, so it prevents any tables or files being
+ created or dropped during that time. See @ref backup and @ref arch_cursor_backup for
+ more information.
+
+All the schema operations listed below perform multi-step metadata modifications. However, they are
+\c non-transactional and will not guarantee ACID properties, see @ref transactions for more
+details. This means that if the application stops abnormally during schema manipulations,
+metadata may be in an invalid state. WiredTiger may or may not be able to recover after this
+type of crash.
+
+@section schema_api Schema Manipulation
+
+All schema manipulations are done in the context of WT_SESSION. All the methods below,
+except WT_SESSION::create and WT_SESSION::truncate, require exclusive access to the specified
+data source(s). If any cursors are open with the specified name(s) or a data source is otherwise
+in use, the call will fail and return EBUSY.
+
+@subsection schema_create Create
+
+The create schema operation is responsible for creating the underlying data
+objects on the filesystem and then creating required entries in the metadata. The API for this
+operation is WT_SESSION::create.
+
+@subsection schema_drop Drop
+
+WT_SESSION::drop operation drops the specified \c uri. The method will delete all related files
+and metadata entries. It is possible to keep the underlying files by specifying
+\c "remove_files=false" in the config string.
+
+@subsection schema_rename Rename
+
+The rename schema operation is responsible for renaming the underlying data object on the
+filesystem and updating the metadata accordingly. See details here WT_SESSION::rename.
```

Comment:
I suggest writing this in the style of drop above in that a uri is given and it will rename all underlying related files.

Reply:
Done